### PR TITLE
Feature/cu k53udz two states machines

### PR DESCRIPTION
**Pull request by JoanMarcBardes**
Explanation of the Two States Machines:

For scripts, it will be always necessary that when we want to use the StateMachineSecondary we go from a state of the StateMachinePrincipal to a state of the Secondary and when we want to stop using the StateMachineSecondary, we must send to the StateMachinSecondary that we want to go from the current state of the Secondary to the current state of the Principal.
In this way the StateMachine will automatically detect that the two StatesMachines are doing the same state and it will stop using the StateMachineSecondary to use only the StateMachinePrincipal.

Ex:
StateMachinePrincipal = Run, StateMachineSecondary = NULL
SendTriggerSecondary(RunAtack);
StateMachinePrincipal = Run, StateMachineSecondary = RunAtack
StateMachinePrincipal = Run, StateMachineSecondary = Atack
OnAnimationSecondaryFinished();
SendTriggerSecondary(AtackRun);
StateMachinePrincipal = Run, StateMachineSecondary = AtackRun
StateMachinePrincipal = Run, StateMachineSecondary = Run
StateMachinePrincipal = Run, StateMachineSecondary = NULL

The Json for StateMachinePrincipal must have only its own States.
    Ex: Idle, Run, Death.
The Json for StateMachineSecondary must have all of the States, the states of the Principal and its own States.
With that we will can interpolate fron StateMachinePrincipal to StateMachineSecondary and backwards.
    Ex: Idel, Run, Death, Atack.

On the other hand,
The Json for StateMachinePrincipal must have all the animated bones of the fbx.
    Ex: CenterNeck, LeftFoot, RightFoot, LeftShoulder, RightShoulder.
The Json for StateMachineSecondary only must have the animated bones of the fbx that will be used.
    Ex: LeftShoulder, RightShoulder.

### CODE REVIEW

**Review by saulzini — DISMISSED** [review]

**Review by JaimeBea — APPROVED** [review]
This is fine as a first approximation for this VS, but it needs some cleanup work:
- There are now too many Manager/Generator/Importer classes. They should be reduced to keep the code as manageable as possible.
- StateMachineManager has functions with way too many parameters. There should be a better  way to pass those parameters, either inside a struct or an existing class.
- StateMachineManager's CalculateAnimation has 10 levels of indentation at some point. This should be reduced either by creating new functions or doing early returns if possible.

File: Project/Source/FileSystem/StateMachinGenerator.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include "Utils/UID.h"
+
+#include <unordered_map>
+
+namespace StateMachinGenerator {
```
Comment (Jordi-Gil) [inline Project/Source/FileSystem/StateMachinGenerator.h:6]: Machine with e

**Review by Jordi-Gil — APPROVED** [review]

**Review by Jordi-Gil — DISMISSED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]
Works fine.

Remove the testing assets and scene.

I will add a small UX issue inside clickUp-QA-UX

**Review by FrancPS — APPROVED** [review]

**Review by saulzini — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by JaimeBea — DISMISSED** [review]

**Review by saulzini — DISMISSED** [review]

**Review by saulzini — DISMISSED** [review]

File: Project/Source/Resources/ResourceClip.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@
 #define JSON_TAG_LOOP "Loop"
 #define JSON_TAG_SPEED "Speed"
 #define JSON_TAG_ID "Id"
-#define JSON_TAG_DURATION_BY_KEYFRAMES_SIZE "DurationByKeyframesSize"
+#define JSON_TAG_FRAMERATE "DurationByKeyframesSize"
```
Comment (JaimeBea) [inline Project/Source/Resources/ResourceClip.cpp:33]: The JSON string is not the same as the tag

Comment (JoanMarcBardes) [reply Project/Source/Resources/ResourceClip.cpp]: Done

**Review by JaimeBea — APPROVED** [review]

**Review by saulzini — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Pull request merged**